src/App/Pages/Settings/LoginPasswordlessRequestsListViewModel.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Bit.App.Abstractions;
+using Bit.App.Resources;
+using Bit.Core.Abstractions;
+using Bit.Core.Models.Response;
+using Bit.Core.Utilities;
+using Xamarin.CommunityToolkit.ObjectModel;
+using Xamarin.Forms;
+
+namespace Bit.App.Pages
+{
+    public class LoginPasswordlessRequestsListViewModel : BaseViewModel
+    {
+        private readonly IAuthService _authService;
+        private readonly IStateService _stateService;
+        private readonly IDeviceActionService _deviceActionService;
+        private readonly IPlatformUtilsService _platformUtilsService;
+        private readonly ILogger _logger;
+        private bool _isRefreshing;
+
+        public LoginPasswordlessRequestsListViewModel()
+        {
+            _authService = ServiceContainer.Resolve<IAuthService>();
+            _stateService = ServiceContainer.Resolve<IStateService>();
+            _deviceActionService = ServiceContainer.Resolve<IDeviceActionService>();
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>();
+            _logger = ServiceContainer.Resolve<ILogger>();
+
+            PageTitle = AppResources.PendingLogInRequests;
+            LoginRequests = new ObservableRangeCollection<PasswordlessLoginResponse>();
+
+            AnswerRequestCommand = new AsyncCommand<PasswordlessLoginResponse>(PasswordlessLoginAsync,
+                onException: ex => HandleException(ex),
+                allowsMultipleExecutions: false);
+
+            DeclineAllRequestsCommand = new AsyncCommand(DeclineAllRequestsAsync,
+                onException: ex => HandleException(ex),
+                allowsMultipleExecutions: false);
+
+            RefreshCommand = new Command(async () => await RefreshAsync());
+        }
+
+        public ICommand RefreshCommand { get; }
+
+        public AsyncCommand<PasswordlessLoginResponse> AnswerRequestCommand { get; }
+
+        public AsyncCommand DeclineAllRequestsCommand { get; }
+
+        public ObservableRangeCollection<PasswordlessLoginResponse> LoginRequests { get; }
+
+        public bool IsRefreshing
+        {
+            get => _isRefreshing;
+            set => SetProperty(ref _isRefreshing, value);
+        }
+
+        public async Task RefreshAsync()
+        {
+            try
+            {
+                IsRefreshing = true;
+                LoginRequests.ReplaceRange(await _authService.GetActivePasswordlessLoginRequestsAsync());
+                if (!LoginRequests.Any())
+                {
+                    Page.Navigation.PopModalAsync().FireAndForget();
+                }
+            }
+            catch (Exception ex)
+            {
+                HandleException(ex);
+            }
+            finally
+            {
+                IsRefreshing = false;
+            }
+        }
```

Comment:
Given that you've change the `Command` that calls this, you are now allowing multiple executions. Have you checked that the control doesn't allow to call this multiple times while refreshing? If the control allows it, then you should check if `IsRefreshing` is `false` in order to proceed with the call to the `authService`
Also this method should be `private`.

Reply:
Cannot check do 
```
if (IsRefreshing)
{
    return;
}
```
Because when using pull to refresh the view already set the variable to `true`.
But it doesn't allow for multiple executions either way.